server/src/main/java/app/stats/service/StatsService.java

```diff
@@ -0,0 +1,224 @@
+        return calculateStreak(activities, false);
+    }
+
+    private Period getCurrentStreak(List<Activity> activities) {
+        return calculateStreak(activities, true);
+    }
+
+    private Duration getTotalTimeThisWeek(List<Activity> activities) {
+        LocalDate startOfWeek = LocalDate.now().with(DayOfWeek.MONDAY);
+        return calculateTotalDuration(activities, startOfWeek.minusDays(1), LocalDate.now());
+    }
+
+    private Duration calculateTotalDuration(List<Activity> activities, LocalDate startDate, LocalDate endDate) {
+        return activities.stream()
+            .filter(activity -> isWithinRange(activity.getDate().toLocalDate(), startDate, endDate))
+            .map(Activity::getDuration)
+            .reduce(Duration.ZERO, Duration::plus);
+    }
+
+    private boolean isWithinRange(LocalDate date, LocalDate start, LocalDate end) {
+        return !date.isBefore(start) && !date.isAfter(end);
+    }
+
+    private String getWeekKey(Activity activity) {
+        int dayOfMonth = activity.getDate().toLocalDate().getDayOfMonth();
+        return "Week " + ((dayOfMonth - 1) / 7 + 1);
+    }
+
+    private void fillMissingWeeks(Map<String, Duration> weeklyDurations, LocalDate lastDayOfMonth) {
+        int totalWeeks = (int) Math.ceil(lastDayOfMonth.getDayOfMonth() / 7.0);
+        for (int i = 1; i <= totalWeeks; i++) {
+            weeklyDurations.putIfAbsent("Week " + i, Duration.ZERO);
+        }
+    }
+
+    private Map<String, Double> convertToHourlyMap(Map<String, Duration> weeklyDurations) {
+        return weeklyDurations.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().toMinutes() / 60.0));
+    }
+
+    private double calculateDailyDuration(List<Activity> activities, LocalDate startOfWeek, DayOfWeek day) {
+        return activities.stream()
+            .filter(activity -> isWithinRange(activity.getDate().toLocalDate(), startOfWeek, startOfWeek.plusDays(6)) &&
+                activity.getDate().toLocalDate().getDayOfWeek() == day)
+            .map(Activity::getDuration)
+            .reduce(Duration.ZERO, Duration::plus)
+            .toMinutes() / 60.0;
+    }
+
+    private Map<String, String> createMostActiveDayResult(DayOfWeek mostActiveDay, Duration totalDuration, List<Activity> activities) {
+        Set<LocalDate> uniqueWeeks = activities.stream()
+            .filter(activity -> activity.getDate().getDayOfWeek() == mostActiveDay)
+            .map(activity -> activity.getDate().toLocalDate().with(DayOfWeek.MONDAY))
+            .collect(Collectors.toSet());
+
+        long weeksCount = uniqueWeeks.size();
+        Duration averageDuration = weeksCount > 0 ? totalDuration.dividedBy(weeksCount) : Duration.ZERO;
+
+        Map<String, String> result = new HashMap<>();
+        result.put("day", timeFormatter.formatDayOfWeek(mostActiveDay));
+        result.put("averageTime", timeFormatter.formatDuration(averageDuration));
+        return result;
```

Comment:
nonono please create an object record and put this info there , dont create map of string string